component/src/test/java/org/wso2/extension/siddhi/execution/ml/classification/perceptron/PerceptronClassifierStreamProcessorExtensionTestCase.java

```diff
@@ -0,0 +1,517 @@
+            inputHandler.send(new Object[]{1.0, 1.0, 0.2, 0.13, "true"});
+            inputHandler.send(new Object[]{0.9, 0.89, 0.3, 0.02, "true"});
+            inputHandler.send(new Object[]{0.0, 0.0, 1.0, 0.82, "false"});
+            inputHandler.send(new Object[]{0.01, 0.4, 0.77, 0.92, "false"});
+            inputHandler.send(new Object[]{0.80, 0.81, 0.11, 0.13, "true"});
+            inputHandler.send(new Object[]{0.02, 0.30, 0.88, 0.76, "false"});
+            inputHandler.send(new Object[]{0.93, 0.71, 0.02, 0.122, "true"});
+            inputHandler.send(new Object[]{0.29, 0.24, 0.98, 0.65, "false"});
+            Thread.sleep(1100);
+
+            InputHandler inputHandler1 = siddhiAppRuntime.getInputHandler("StreamA");
+            // send some unseen data for prediction
+            inputHandler1.send(new Object[]{0.8, 0.67, 0.1, 0.03});
+            inputHandler1.send(new Object[]{0.33, 0.23, 0.632, 0.992});
+            Thread.sleep(1000);
+            // try to drift the model
+            inputHandler.send(new Object[]{0.88, 1.0, 0.2, 0.13, "false"});
+            inputHandler.send(new Object[]{0.9, 0.79, 0.3, 0.02, "false"});
+            inputHandler.send(new Object[]{0.0, 0.0, 0.90, 0.72, "true"});
+            inputHandler.send(new Object[]{0.01, 0.4, 0.87, 0.62, "true"});
+            inputHandler.send(new Object[]{0.90, 0.91, 0.11, 0.13, "false"});
+            inputHandler.send(new Object[]{0.02, 0.30, 0.88, 0.66, "true"});
+            inputHandler.send(new Object[]{0.83, 0.79, 0.02, 0.122, "false"});
+            inputHandler.send(new Object[]{0.29, 0.24, 0.98, 0.77, "true"});
+            Thread.sleep(1000);
+            // send some unseen data for prediction
+            inputHandler1.send(new Object[]{0.8, 0.67, 0.1, 0.03});
+            inputHandler1.send(new Object[]{0.33, 0.23, 0.632, 0.992});
+
+            AssertJUnit.assertEquals(4, count);
+            Thread.sleep(1100);
+        } catch (Exception e) {
+            logger.error(e.getMessage());
+        } finally {
+            siddhiAppRuntime.shutdown();
+        }
+    }
+
+    @Test
+    public void testClassificationStreamProcessorExtension2() throws InterruptedException {
+        logger.info("PerceptronClassifierUpdaterStreamProcessorExtension TestCase - Features are not of type double");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "define stream StreamA (attribute_0 double, attribute_1 double, attribute_2 " +
+                "double, attribute_3 bool, attribute_4 string );";
+        String query = ("@info(name = 'query1') from StreamA#streamingml:perceptronClassifier('model1', " + "0.0,0.5," +
+                " attribute_0, attribute_1, attribute_2, attribute_3) \n" + "insert all events into outputStream;");
+
+        try {
+            SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(inStreamDefinition + query);
+            AssertJUnit.fail();
```

Comment:
Why are we asserting fail. Are we expecting an exception? If so please use expected exceptions. Same for testcases below

Reply:
We are expecting a exception and then we are asserting the exception messages.